packages/backend-common/src/context/RootContext.ts

```diff
@@ -14,29 +14,36 @@
  * limitations under the License.
  */
 
-import { AbortSignal } from 'node-abort-controller';
 import { Context } from './types';
 
 /**
- * Since the root context can never abort, and since nobody is every meant to
- * dispatch events through it, we can use a static dummy instance for
+ * Since the root context can never abort, and since nobody is ever meant to
+ * dispatch events through it, we can use a static fake instance for
  * efficiency.
  */
-const dummyAbortSignal: AbortSignal = Object.freeze({
-  aborted: false,
-  addEventListener() {},
-  removeEventListener() {},
-  dispatchEvent() {
-    return true;
-  },
-  onabort: null,
-});
+const fakeAbortController = new AbortController();
+const fakeAbortSignal: AbortSignal = Object.freeze(
+  Object.assign(
+    {
+      aborted: false,
+      reason: undefined,
+      throwIfAborted() {},
+      onabort() {},
+      addEventListener() {},
+      removeEventListener() {},
+      dispatchEvent() {
+        return true;
+      },
+    },
+    fakeAbortController.signal,
+  ),
+);
```

Comment:
This is one place where it was clear that there were some very minor differences between them. Performing this dance instead of making a full fake seemed like the easiest way of making one where the prototype was preserved etc. Happy to hear if there are other better ways :)

Reply:
A potentially cleaner way is to create a `class FakeAbortSignal extends AbortSignal { ... }` instead.

Otherwise I'm feeling it's better to flip this assignment around, either way should work, but feels a bit more natural to be overriding the members on the signal instead right?